src/main.rs

```diff
@@ -22,6 +22,9 @@ mod worker;
 // use smol::prelude::*;
 use crate::worker::{Worker, WorkerConfig};
 
+// get the version of this program itself (avoid use 'env!' because it will crashes if the program is compiled without cargo)
+const VERSION: Option<&str> = option_env!("CARGO_PKG_VERSION");
```

Comment:
This is unnecessary and unidiomatic. Nobody compiles Rust without Cargo these days and the vast majority of the Rust ecosystem depends on Cargo being around. Most crates cannot be compiled without Cargo.